src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -125,60 +126,139 @@ class CoreWorkerDirectActorTaskSubmitter {
   friend class CoreWorkerTest;
 };
 
+/// Object dependency and RPC state of an inbound request.
+class InboundRequest {
+ public:
+  InboundRequest(){};
+  InboundRequest(std::function<void()> accept_callback,
+                 std::function<void()> reject_callback, bool has_dependencies)
+      : accept_callback_(accept_callback),
+        reject_callback_(reject_callback),
+        has_pending_dependencies_(has_dependencies) {}
+
+  void Accept() { accept_callback_(); }
+  void Cancel() { reject_callback_(); }
+  bool CanExecute() { return !has_pending_dependencies_; }
+  void MarkDependenciesSatisfied() { has_pending_dependencies_ = false; }
+
+ private:
+  std::function<void()> accept_callback_;
+  std::function<void()> reject_callback_;
```

Comment:
const

Reply:
Seems marking this as const prevents copy construction...